proxy/http2/Http2ConnectionState.cc

```diff
@@ -1242,13 +1242,8 @@ Http2ConnectionState::release_stream(Http2Stream *stream)
     --total_client_streams_count;
   }
 
-  if (ua_session) {
-    SCOPED_MUTEX_LOCK(lock, this->ua_session->mutex, this_ethread());
-    if (!ua_session) {
-      // Workaround fix for GitHub #4504. The `ua_session` could be freed while waiting for acquiring the above lock.
-      return;
-    }
-
+  SCOPED_MUTEX_LOCK(lock, this->mutex, this_ethread());
+  if (this->ua_session) {
```

Comment:
I wonder if you want an `ink_assert` here to guarantee `this->mutex == ua_session->mutex`. At least a comment to that effect.

Reply:
Added the` ink_assert`.